crates/turborepo-filewatch/src/package_watcher.rs

```diff
@@ -399,7 +404,19 @@ impl<T: PackageDiscovery + Send + Sync + 'static> Subscriber<T> {
                 self.handle_package_json_change(file_event).await
             }
             Err(()) => Err(()),
-        }
+        };
+
+        // now that we have updated the state, we should bump the cookies so that
+        // people waiting on downstream cookie watchers can get the new state
+        self.cookie_tx.register(
```

Comment:
The idea is that you have two entites (glob watch and package watch) whose 'processing index' may not always be in sync.

The package hash watcher will also be processing file updates, so it will also need its own independent tracking mechanism, but to produce a valid answer to a request at a particular point of time you need to ensure that both the cookied data in that hash watcher _and_ in the package watcher are at least that far along, otherwise you may be using a package hash that  is up to date as far as it is concerned, but is using stale data from package discovery. I will add this nuance in the `cookie.rs` docs.

Reply:
Sorry, what do you mean by "processing index"? Like which file events have been seen?

Reply:
Yeah, one actor (A) may have processed 100 but then is currently doing some recalculation, while another (B) has processed 200. A request comes in at index 200 which depends on file system data, so we wait for a cookie which (assuming no changes between) is event 201. If actor B has a dependency on the data in actor A, then a request to actor B should wait for actor A to update its data to at least 201 before it can do the read.

Reply:
Ahh gotcha, that makes sense, thanks